scripts/util.py

```diff
@@ -701,6 +701,37 @@ def cached_property(f):
     return property(lru_cache()(f))
 
 
+def retry(max_retries: int, init_wait_time: float, warn_msg: str, exc_type: Exception):
+    """Retries functions that raises the exception exc_type.
+    Retry time is increased by a factor of two for every iteration.
+
+    Args:
+        max_retries (int): Maximum number of retries
+        init_wait_time (float): Initial wait time in secs
+        warn_msg (str): Message to print during retries
+        exc_type (Exception): Exception type to check for
+    """
+
+    def decorator(f):
+        def wrapper(*args, **kwargs):
```

Comment:
nit. consider adding `functools.wraps` for cleaner runtime

Reply:
Done